app/include/zmk/behavior.h

```diff
@@ -8,12 +8,12 @@
 
 struct zmk_behavior_binding {
     char *behavior_dev;
-    u32_t param1;
-    u32_t param2;
+    uint32_t param1;
```

Comment:
I believe they added a Kconfig setting to enable backwards compatible types for this.

Also refactoring these means while we work on this, we can't merged other fixes due to more conflicts.

Thoughts on doing then types changes after the initial 2.4.0 merge?

Reply:
Could do.  I prefer ripping off the band-aid personally.

The existing PRs will need to be rebased regardless because 2.4 has many dependency changes and other non-optional API changes that will conflict (anything involving the device API).  So from that perspective there's wisdom in requiring blanket rebasing and testing.  That said, I'm not opposed to a graduated approach if that's something you'd like to manage.  It has the potential to drag on and on though.

I vote for a line in the sand, it's up to you though.

Reply:
I'm not sure where you are on this, but it's probably worth highlighting that it's the second commit (and largest).  6 of the subsequent commits build on top of it.  So I'd effectively have to redo the bulk of this PR again and we'd have to restart testing.

Reply:
I'm fine keeping this as part of this. Thanks for circling back.